lib/tools/apk-utils.js

```diff
@@ -489,11 +489,42 @@ apkUtilsMethods.setDeviceLocale = async function (locale) {
   await this.setDeviceLanguageCountry(split_locale[0], split_locale[1]);
 };
 
+/**
+ * Make sure current device locale is expected or not.
+ *
+ * @param {string} language - Language. The language field is case insensitive, but Locale always canonicalizes to lower case.
+ * @param {string} country - Country. The language field is case insensitive, but Locale always canonicalizes to lower case.
+ *
+ * @return {boolean} If current locale is language and country as arguments, return true.
+ */
+apkUtilsMethods.ensureCurrentLocale = async function (language, country) {
+  if (!_.isString(language) && !_.isString(country)) {
+    log.warn('validateLocale requires language or country');
+    return false;
+  }
+
+  if (await this.getApiLevel() < 23) {
+    const curLanguage = await this.getDeviceLanguage();
```

Comment:
Is it guaranteed that these methods (`getDeviceLanguage`, `getDeviceCountry`, `getDeviceLocale`) will always return a string (there is no chance of null or undefined)? If there is _any_ chance then it should be guarded against.

Reply:
Yes, it returns `""` or `"en"`(some variations). So, I haven't implemented guard.